### PR TITLE
chore(observability): deprecate obsolete http metrics

### PR DESCRIPTION
**Pull request by dsmith3197**

### CODE REVIEW

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md [hunk 1]
```diff
@@ -31,3 +35,15 @@ been removed from the Enterprise configuration. Instead of `region`, `site` shou
 #### Removal of Deprecated `component_name` Metric Tag {#deprecated-component-name}
 
 The deprecated `component_name` tag has been removed from all internal metrics. Instead the `component_id` tag should be used.
+
+### Deprecations
+
+#### Deprecation of `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` Internal Metrics {#deprcate-obsolete-http-metrics}
+
+The `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` internal metrics will be removed in the 0.35.0 release.
```
Comment (jszwedko) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:43]: Will they be replaced with anything? Or we just don't think they are valuable?

Comment (dsmith3197) [reply website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:43]: Yes, the replacements are provided in the table below.

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md [hunk 2]
```diff
@@ -14,6 +14,10 @@ Vector's 0.34.0 release includes **breaking changes**:
 1. [Removal of Deprecated Datadog Component Config Options](#datadog-deprecated-config-options)
 1. [Removal of Deprecated `component_name` Metric Tag](#deprecated-component-name)
 
+and **deprecations**:
+
+1. [Deprecation of `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` Internal Metrics](#deprcate-obsolete-http-metrics)
```
Comment (urseberry) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:19]:
```suggestion
1. [Deprecation of `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` Internal Metrics](#deprecate-obsolete-http-metrics)
```

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md [hunk 3]
```diff
@@ -31,3 +35,15 @@ been removed from the Enterprise configuration. Instead of `region`, `site` shou
 #### Removal of Deprecated `component_name` Metric Tag {#deprecated-component-name}
 
 The deprecated `component_name` tag has been removed from all internal metrics. Instead the `component_id` tag should be used.
+
+### Deprecations
+
+#### Deprecation of `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` Internal Metrics {#deprcate-obsolete-http-metrics}
```
Comment (urseberry) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:41]:
```suggestion
#### Deprecation of `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` Internal Metrics {#deprecate-obsolete-http-metrics}
```

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md (hunk 1 above)
Comment (urseberry) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:43]:
```suggestion
The `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` internal metrics are removed in the 0.35.0 release.
```
The public docs style guide calls for all docs to be written in present tense.

File: website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md [hunk 4]
```diff
@@ -31,3 +35,15 @@ been removed from the Enterprise configuration. Instead of `region`, `site` shou
...
+
+The `requests_completed_total`, `request_duration_seconds`, and `requests_received_total` internal metrics will be removed in the 0.35.0 release.
+
+| Deprecated Metric           | Replacement Metric                  | Affected Components                                                                      |
```
Comment (urseberry) [inline website/content/en/highlights/2023-11-07-0-34-0-upgrade-guide.md:45]:
```suggestion
| Deprecated metric           | Replacement metric                  | Affected components                                                                      |
```

**Review by urseberry — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]

**Pull request merged**